hs-bindgen/src-internal/HsBindgen/Backend/Hs/Translation.hs

```diff
@@ -1047,8 +1053,12 @@ typ' ctx = go ctx
     go _ (C.TypePrim p) =
         Hs.HsPrimType (goPrim p)
     go _ (C.TypePointer t) = case t of
-        C.TypeFun {} -> Hs.HsFunPtr (go CPtrArg t)
-        _            -> Hs.HsPtr (go CPtrArg t)
+        C.TypeTypedef (C.TypedefRegular C.NamePair { nameC })
+          | Just C.TypeFun{} <- Map.lookup nameC typedefs
+                      -> Hs.HsFunPtr (go CPtrArg t)
+          | otherwise -> Hs.HsPtr (go CPtrArg t)
+        C.TypeFun {}  -> Hs.HsFunPtr (go CPtrArg t)
+        _             -> Hs.HsPtr (go CPtrArg t)
```

Comment:
I think this way of looking through typedefs is fine -- it's also one of the options I considered while working on #1035 and #1033. I might have a slightly different approach that I want to experiment with, but I'm not yet sure if it is going to work, so IMO this is good as is!

Reply:
True, I overlooked recursive unfolding. That would still be good to include

Reply:
No, the `Map` _itself_ fine, it just gives definitions.  We need Joris' proposed "normalize".

Reply:
I don't have a written down proposal just yet

Reply:
Btw, Maybe this bit of code should be using `getUnderlyingType:

```suggestion
go _ (C.TypePointer t) = case getUnderlyingType typedefs t of
        C.TypeFun {}  -> Hs.HsFunPtr (go CPtrArg t)
        _             -> Hs.HsPtr (go CPtrArg t)
```